### DISCUSSION TITLE
Native Apache Avro support for requests

### DISCUSSION THREAD

**Discussion by SpOOnman** [Kong/insomnia#3598] [Ideas]
**Is your feature request related to a problem? Please describe.**
Insomnia lacks Apache Avro support.

**Describe the solution you'd like**
I want to work on Apache Avro support in Insomnia. I think that this may be close to GraphQL editor. There will be two editable editors (in one panel?): one for query and one for schema. I will use official JS avro library to check if a query matches the schema and show warnings if they don't. When they match user can send a request with a binary content that's encoded with avro library on-the-fly.

**Describe alternatives you've considered**
I didn't find any REST client with avro support. Without a tool I have to use some CLI tools, write binary to a file and send content from a file, which is cumbersome.

**Additional context**
I think I can work on this on a hackaton alone or with my colleagues.

**Comment by SpOOnman** [comment]
Can you please mark it with hacktoberfest label? Thanks!

**Comment by andrzejgulak** [comment]
Hey, has there been any progress?
I'd love to see this feature :)

**Comment by dimitropoulos** [comment]
It's been a year, so I'm curious.  Well, haha, I'm always going to be curious no matter the timeframe:

Does anyone know of a similar tool that supports Avro?

**Reply by SpOOnman** [reply]
I don't know any other tool that supports Avro. I did some initial work for a few hours but I've dropped it.
